sympy/core/power.py

```diff
@@ -969,7 +969,11 @@ def _eval_is_rational(self):
             return False
         if e.is_integer:
             if b.is_rational:
-                return True
+                if b.is_nonzero or e.is_nonnegative:
+                    return True
+                # In the special case of b**b, the result is rational
```

Comment:
Why do you think it does explain anything?  This is just verbose equivalent of the following lines.

Perhaps, you want to explain why case `b**b` is rational.